### ISSUE TITLE
Account data transfer between iOS devices extremely slow

### ISSUE THREAD

**Issue by neuhaus** [signalapp/Signal-iOS#5647]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
The transfer of Signal account between two iOS devices takes too long.

### Steps to reproduce
- Initiate account transfer between two iOS devices
- transfer takes several hours
- when someones calls you, the transfer gets interrupted and has to be restarted.

**Actual result:** Describe here what happens after you run the steps above (i.e. the buggy behaviour)
The transfer takes several hours.

**Expected result:** 
The transfer should be adequately fast.
I have 7 GB of Signal data. I have Wifi with more than 200mbit/s speed. Even if you divice it by half it should take less than 10 minutes instead of 3+ hours.

### Device info

**Device**: iPhone 13 and iPhone SE 2020

**iOS version**: 16.6

**Signal version:** 6.36

### Link to debug log
n/a

**Comment by philmtd** [comment]
Same issue here. In the end, I resigned and just set it up without a backup, losing all message and media history. Super frustrating experience.

**Comment by Miguel-Signal** [comment]
Hi @neuhaus 

If you are still experiencing this could you reach out to support@signal.org and reference this Github issue?

**Comment by dulnan** [comment]
I'm also experiencing this right now. Transfer between old and new iPhone is running since 4 hours and is currently at 73%.

I can understand that it might take long. For me personally it would have helped if there was a rough estimate at the beginning. If I knew it'd take this long I would have done it overnight.

**Comment by karbak** [comment]
Just another data point - ~4 GB of data took just over an hour to transfer between the current version of Signal running on an iPhone 11 (iOS 16.6) and iPhone 15 (iOS 17.0.2) sitting right next to each other on the same Wifi 6 wireless network with the access point in the same room. The first time I tried this, when I clicked on a notification on one of the phones with about 50% of the transfer completed, the transfer aborted, and I was forced to start over.

**Comment by skalarproduktraum** [comment]
I can confirm this behaviour as well, when transferring data between an iPhone 12 and an iPhone 15 Pro, both running iOS 17.0.2. The amount of data to be transferred is about 10 GiB, for which Signal anticipates 3h. The transfer already got interrupted once at 50% and had to be started over. 

The user experience would be tremendously better if:
* transfers could be resumed
* as most people are likely running this on a phone, it'd be great if taking a call would not immediately cancel the transfer. A notification to switch back to Signal would be helpful if the transfer is allowed to run in the background for a limited amount of time

**Comment by haloeight** [comment]
+1 to this problem. Transferring from an iPhone 14 PM to 15 PM, the transfer time goes for 3-4 hours and then fails between 39-67%, requiring application or device restart, failing again. So right now this is blocking me from using my phone for over a day. Even with the encryption overhead Apple does manage to transfer all other device data / configuration faster than Signal does.

**Comment by MalEbenSo** [comment]
+1. My transfer took 2.5h for ca. 10 GB

Transfer had good conditions: Modern iPhones with direct access to WiFi, close proximities, no obstacles, no disturbance from neighbors, Wifi in other instances uses 500-700 Mbit/s.

Unless I screwed my math, the above transfer is 4 GB/h or 32 Gbit/h or around 9 Mbit/s. Ouch!

**Comment by chingham** [comment]
+1 So damn slow...

Considering the speed of my network, the fact that both phones are quite recent (iPhone 13 Pro -> iPhone 15 Pro), and the data transfered, I would expect the process to be maximum 2 minutes long, not 2 hours...

This is really raging, especially since we cannot do anything else during the transfer with both phones, not even switch app or accidently lock the screen... Or the transfer is simply cancelled and we have to start all over again !

**Comment by Flexusma** [comment]
Same issue here, waiting about 1h for the transfer to complete (2-4gb chat size) (iphone 12->14) both using the latest version of signal

**Comment by hoechenberger** [comment]
I just ran into this issue moving from my iPhone XR to a 13 Pro. It's been running for 1h and estimates it will take another 10, for about 10 GB of data. In the beginning I accidentally locked the phone and the transfer was aborted. It's an extremely frustrating experience.

**Comment by vinnyoflegend** [comment]
I'm not sure if this will help anyone as I was seeing similar issues as reported here even though my chat data is insignificant at about 500MB.

I was transferring from an iPhone SE2 to SE3 (both on IOS 17.1.x) Signal was estimating 16-30 minutes but failing to move past 0% most of the time. I tried device restarts, quitting apps, flipping airplane mode on/off, etc.

On one attempt where I was able to kick it off, it got to 99% in 20 minutes and never completed after waiting 30 minutes.

Just now, I kicked off an attempt where I directly laid one phone on top of the other as soon as it presented the TX/RX screens on each. Laying the devices on top of each other triggered the NFC/Airdrop linking (in case that's relevant). Progress started and fully completed within 3 minutes.

**Comment by st4tic-rr** [comment]
> I'm not sure if this will help anyone as I was seeing similar issues as reported here even though my chat data is insignificant at about 500MB.
> 
> I was transferring from an iPhone SE2 to SE3 (both on IOS 17.1.x) Signal was estimating 16-30 minutes but failing to move past 0% most of the time. I tried device restarts, quitting apps, flipping airplane mode on/off, etc.
> 
> On one attempt where I was able to kick it off, it got to 99% in 20 minutes and never completed after waiting 30 minutes.
> 
> Just now, I kicked off an attempt where I directly laid one phone on top of the other as soon as it presented the TX/RX screens on each. Laying the devices on top of each other triggered the NFC/Airdrop linking (in case that's relevant). Progress started and fully completed within 3 minutes.

Can confirm, placing the two phones directly on top of each other makes the transfer substantially faster. I did this half way through mine and the estimated time halved from 1+ hours to 40 or so minutes for a large amount of data. Would be nice to see this included in Signal's documentation for the transfer process. Thanks for the advice @vinnyoflegend! Just saved me another hour or so of waiting :)

**Comment by notpeter** [comment]
An additional anecdotal piece of data: Transferring history between two iPhone SE3 devices was unbearably slow and after multiple attempts did not succeeded.  This is especially problematic because this wasn't a device upgrade, where I had phones at home and could just leave them plugged in overnight. It was a hardware replacement (button failure) so I've been sitting in an Apple store for many hours waiting trying to transfer my data. After more than 4 hours and two interrupted transfers I gave up and was forced to abandon my message history.

I think it's probably worth creating some additional issues in addition to this one about "slowness".

- [ ] Investigate causes of transfer failure
- [ ] Feature: resumable data transfer
- [ ] Feature: move between phones without transferring message history.

I don't know if Signal Foundation does bounties, but after wasting many hours and ultimately triggering a safety number change with all my contacts and group threads, I would personally fund work on this so others have a better experience.

**Comment by zubko** [comment]
Super slow transfer as well. 

As for me I don't care about transferring images or photos - it would be great to be able to choose to transfer just chats and the text history.

**Comment by Orava2** [comment]
> +1. My transfer took 2.5h for ca. 10 GB
> 
> Transfer had good conditions: Modern iPhones with direct access to WiFi, close proximities, no obstacles, no disturbance from neighbors, Wifi in other instances uses 500-700 Mbit/s.
> 
> Unless I screwed my math, the above transfer is 4 GB/h or 32 Gbit/h or around 9 Mbit/s. Ouch!

I'm currently transferring data. I'm at 5 % and ETA is 4 hours. My WiFi access point shows 0 Mbps so I think that the data is transferred over Bluetooth.

**Comment by sgreadly** [comment]
You're all lucky. I had about 30GB+ to transfer due to some bug that doesn't delete older media. It wanted a few days to transfer. A few days where I have to keep both phones next to each other and not touch or use them else it fails. 

Cat jumped and somehow closed the screen after a day. I gave up.

**Comment by anoxi** [comment]
> I'm not sure if this will help anyone as I was seeing similar issues as reported here even though my chat data is insignificant at about 500MB.
> 
> I was transferring from an iPhone SE2 to SE3 (both on IOS 17.1.x) Signal was estimating 16-30 minutes but failing to move past 0% most of the time. I tried device restarts, quitting apps, flipping airplane mode on/off, etc.
> 
> On one attempt where I was able to kick it off, it got to 99% in 20 minutes and never completed after waiting 30 minutes.
> 
> Just now, I kicked off an attempt where I directly laid one phone on top of the other as soon as it presented the TX/RX screens on each. Laying the devices on top of each other triggered the NFC/Airdrop linking (in case that's relevant). Progress started and fully completed within 3 minutes.

Don’t know if this helped much since the transfer still takes about an hour for me.

Anyone from signal looking into this?

I guess a lot of people just give up during the long transfer, start with a new profile and accept the data loss, which is a sad experience on iOS.

**Comment by chingham** [comment]
Tbh I really think nobody at signal is even interested
Almost a year since this issue has been opened and not a single answer explaining why or giving a solution
Signal is updated once a week and yet real issues are never addressed…

**Comment by MalEbenSo** [comment]
I got my new iPad and – sadly as expected – the transfer of Signal messages took longer than the transfer of everything else. 

Around 13 hours for 13 GB. 
Unless my math fails me 1 GB/hour translates to 291 KB/s or 2330 Mb/s. 

The devices were right next to each other. I also tried to stack them on top of each other, as somebody mentioned that this might speed things up by using NFC communication. 

Frankly, I should not have to transfer data at all. I want to keep data in my iCloud account, one data store for all devices, automatically in sync. That's for iOS, but other cloud storage should be possible, e. g. company private clouds for professional users.
I am happy with (i. e. I trust) Apple's so-called "Advanced Data Protection".
Optionally Signal could offer to encrypt the data store. But really optionally, because Signal's E2EE-responsibility ends once I can see, copy, download messages. 
Then the whole issue of slow data transfer becomes irrelevant.

**Comment by zer777** [comment]
This is going to be very relevant again as soon as the first iPhone 16 are shipped to customers :D.

One of my top reasons not to switch iPhones is to avoid this issue. 
So in a funny way, Signal transfer being too slow makes us a bit more eco friendly 😀

**Comment by neuhaus** [comment]
This issue is still relevant.

**Comment by Jamo33** [comment]
Last time i updated my iPhone it took over 24 hours to transfer signal. That's more than 24 hours that I couldn't use my phone. Completely unacceptable. 

I will be updating my phone again in the coming months and I am not looking forward to this process. 

My brother recently updated his phone and he couldn't wait the time it would take him so he had to wipe it all, resulting in all the conversations that he had with our soon to be dead father being deleted. A very very sad thing. 😔

I would also like to point out that because it takes so long you have to prevent the screen from closing on both devices. I know this from experience but many others would have no idea and their resulting experience is crap.

**Comment by sgreadly** [comment]
@Jamo33, for what it's worth, I did a migration earlier this year. I can't remember the size of my Signal, maybe around 5-7GB, but it didn't take long. Maybe an hour or so, which was surprising since the last time I did it ~2+ years ago it wanted 2 days.

That said, it does help if you go through your media and delete some old stuff you don't need anymore. I do that with memes, especially video memes and notice a significant difference in used storage.

**Comment by MalEbenSo** [comment]
>  it does help if you go through your media and delete some old stuff you don't need anymore.

That's obvious. It doesn't solve the underlying problem, that the transfer is taking too long. Like _way_ too long.

**Comment by jzielke84** [comment]
The devs of Signal have become like the ones of Jira. Implementing everything but not caring about essential core stuff like a useful ios backup without having a second phone at hand. And even if you have, It's so god damn slow it's just ridiculous...

**Comment by Intrepidd** [comment]
Currently transferring my data to the new iPhone 16, 3-4 hours of transfer are estimated. This is ridiculous, and I'm already expecting the process to fail one or more times because of random issues or one of the phones going to sleep.

For comparison, it took me 40 seconds to sync my whatsapp messages, then it got the media in the background in a few minutes.

**Comment by lukescammell** [comment]
This is one of those things that really makes it difficult to be a proponent for Signal and convince other people to use it. @jzielke84 os absolutely right about the devs ignoring core functions while adding frivolities.

**Comment by notpeter** [comment]
I haven't confirmed this, but I think the issue may be exacerbated by network conditions.  Specifically if you are in an environment (e.g. at an Apple store getting an new iPhone) where wireless isolation is enabled (wireless clients can't talk to each other) I think the transfer may actually happen over bluetooth instead of via WiFi.  There is no indication of this other than the lack of network traffic.

Hilariously I was able to work around this by using a wireless hotspot. I connected both phones to the wireless hotspot and instead of the many hour estimate I got on the Apple Wifi everything transferred in ~40mins.

Perhaps this request should be split into two distinct enhancement requests:
1. An indicator whether WiFi or bluetooth is being used for the transfer and warning that bluetooth will be slow.
2. Support for iOS Direct Peer Wifi connections via [Multipeer Connectivity Framework](https://developer.apple.com/documentation/multipeerconnectivity).

**Comment by jzielke84** [comment]
I was having the issue at home with 802.11ac Ubiquiti based equipment. Enough bandwidth to sync everything. No need for direct peer APs.

**Comment by notpeter** [comment]
> I was having the issue at home with 802.11ac Ubiquiti based equipment. Enough bandwidth to sync everything. No need for direct peer APs.

Not saying you have this enabled, but even at home "Client Device Isolation" or assigning two devices to isolated VLANs are easy enough to mis-configure.
<img width="855" alt="Screenshot 2024-09-20 at 12 56 25" src="https://github.com/user-attachments/assets/05004459-a888-42d2-a0cf-14059e8a06f5">

Curious whether you observed WiFi network usage from these devices during the transfer or if it was using something else (Bluetooth) as is my conjecture.

**Comment by jzielke84** [comment]
Sure but no such thing in my case. So it’s just badly implemented.

**Comment by sneakerdad** [comment]
would be great to have an option for transfer over cable (usb-c to usb-c in case of iPhone 15 Pro -> iPhone 16 Pro)

**Comment by nickmccollum** [comment]
This issue combined with iMessages support for E2E has made me reconsider my usage of Signal and thus my continuing contributions over the many years.

**Comment by TopBanana123** [comment]
Same issue here, going from my iPhone 11 to my 16 Pro and the estimate is 5 hours and I must keep the app open, pretty crazy

**Comment by rahearn** [comment]
I just got a new phone and have a multiple hours transfer in progress. I can't understand how it's possibly this slow

**Comment by FxRe** [comment]
It appears they are working on an entirely new backup system (https://signalupdateinfo.com/news/cloud-backups.html), maybe they’re focusing entirely on that project and don’t spend time fixing/improving the current backup implementation. But the lack of communication from the Signal team in here is disappointing indeed.

**Comment by TopBanana123** [comment]
> It appears they are working on an entirely new backup system (https://signalupdateinfo.com/news/cloud-backups.html), maybe they’re focusing entirely on that project and don’t spend time fixing/improving the current backup implementation. But the lack of communication from the Signal team in here is disappointing indeed.

Sounds okay but I don't really care about a backup, I wouldn't bother to turn it on and wouldn't pay for it. I feel this this is a completely separate issue and feature

**Comment by amanda-walker** [comment]
Trying to transfer from an iPhone 13 Pro Max to an iPhone 16 Pro Max, the QR stuff worked but it's just stuck at 0%.  No estimate, no apparent progress for half an hour.

**Comment by astubbs** [comment]
Aside from also taking an insanely long time for this, for me signal would simply crash at the qr code part when you click “don’t see anything on your old phone?” Unless I manually turned on network access for signal in settings.

**Comment by leedrake5** [comment]
Just wanted to drop a note that this is an active issue. Got stuck at 99% with an estimated 1 second to completion for about an hour before giving up.

**Comment by toksdotdev** [comment]
I was also running into the work until 99%, but never complete issue. I realized that I was typing the wrong pin at the beginning of the transfer process.

Apparently the transfer starts happening even though you input the wrong pin at the beginning, but this is never communicated to the user (you can type the wrong pin to replicate this behavior).

I didn't realize this until after 2 tries. I'll recommend verifying the pin you entered at the beginning. You can also change pin before the transfer, and use your most recent pin—perhaps you've forgotten what it is.

It does however still take an unusual amount of time to complete the transfer as others have mentioned.

**Comment by MalEbenSo** [comment]
> I haven't confirmed this, but I think the issue may be exacerbated by network conditions. […] 
> 
> Perhaps this request should be split into two distinct enhancement requests:
> 
> 1. An indicator whether WiFi or bluetooth is being used for the transfer and warning that bluetooth will be slow.
> 2. Support for iOS Direct Peer Wifi connections via [Multipeer Connectivity Framework](https://developer.apple.com/documentation/multipeerconnectivity).

Of course network conditions can exacerbate any network transfer, so this is always true. 
The point here is that this transfer problem …
1. appears to be rather common, 
2. suffers from abysmally slow transfers,
3. occurs under otherwise good networking conditions,
4. is exacerbated by requirements to keep app in front, phones unlocked,
5. is flaky and may fail after hours of process,
6. equates to data loss on a new phone if transfer is ultimately impossible (in app and out of the app for lack of export / backup options.)

Bringing up Bluetooth triggered some new ideas in me: Wifi and Bluetooth are ([or were?](https://www.tomshardware.com/reviews/local-area-network-wi-fi-wireless,3020-6.html)) known to interfere with each other. 
Thus has anybody tried to transfer …
1. with Bluetooth turned off or
2. with a _larger_ distance between the two devices? (I always kept my old and new phone really close, expecting to enable a better connection, but maybe I caused the opposite?)

**Comment by MalEbenSo** [comment]
> This issue combined with iMessages support for E2E has made me reconsider my usage of Signal and thus my continuing contributions over the many years.

Same here. I am a regular monthly supporter, and while I do not proselytize (I think 😅), I have explained to others upon request why Signal is my cross-platform messenger of choice, and they have installed it after that to communicate with me. 

But I find it hard to recommend Signal if data are locked in the app, there is no backup, and transfer to other devices is so problematic. 

This is not a cosmetic issue or some minor nuisance. This is at the very least a severe usability issue in a common use case (outside of messaging) and can mean data loss in some cases.

**Comment by MalEbenSo** [comment]
> It appears they are working on an entirely new backup system (https://signalupdateinfo.com/news/cloud-backups.html), maybe they’re focusing entirely on that project and don’t spend time fixing/improving the current backup implementation. But the lack of communication from the Signal team in here is disappointing indeed.

Indeed a good cloud solution could solve various current problems, including this one:
1. Slow transfer speeds and possibly data loss.
2. Lacking backup in the case of device loss. 
3. Syncing history to all devices.
4. Lacking user sovereignty over user data (i. e. export option and no vendor lock-in).

By comparison: 
- For iMessage I setup any new device for the Apple ID that is used. 
- Transfer to new device or adding new device is simple and fast. 
- Loss of one device (damage, theft, ...) is not an issue. 
- With ADP (Advanced Data Protection) Apple claims they cannot see user data.
- For the paranoid: I can also check which devices are attached to my ID. 

The Apple iMessage way 
- does _not_ solve problem 4.,
- is not cross platform and
- is not open source.

Always hoping for the best from the Signal Team I would hope that Signal's cloud solution …
- works similar to the above _plus_
- allows users to choose from various common cloud services
- is encrypted in the cloud by Signal (i. e. does not rely cloud provider encryption)
- is open source
- is cross platform

**Comment by neuhaus** [comment]
Personally, i don't want my database dump uploaded to a cloud, even if encrypted. I prefer a solution for iOS as implemented for Android.

**Comment by axelvugts** [comment]
For anyone else experiencing extremely slow Signal data transfers, the following improved my tranfer time by 4x.
I have reasonably fast WiFi 6 at home (800-1200Mbps) but still the transfer of my small amount of Signal data was estimated to take over 2 hours. Then I connected my new phone (iPhone 16 Pro) to the network using an ethernet cable and a USB-C to Ethernet adapter. The phone switched from WiFi to Ethernet mid-transfer and the estimated time dropped to ~30 mins.

**Comment by Jamo33** [comment]
> For anyone else experiencing extremely slow Signal data transfers, the following improved my tranfer time by 4x.
> 

Did you have Bluetooth turned on by any chance? I'm just wondering if for some reason it's trying to use Bluetooth rather than WiFi. 

Definitely not looking forward to getting my new iPhone. 🫣

**Comment by sgreadly** [comment]
@axelvugts , Can you recommend a good USB-C Ethernet adapter that works with the iPhone? Preferably one on Amazon? =D

**Comment by Jamo33** [comment]
> Then I connected my new phone (iPhone 16 Pro) to the network using an ethernet cable and a USB-C to Ethernet adapter.
> 


You only connected the receiving phone to the Ethernet? I wonder what would happen if you connected both?

**Comment by MalEbenSo** [comment]
> > For anyone else experiencing extremely slow Signal data transfers, the following improved my tranfer time by 4x.
> 
> Did you have Bluetooth turned on by any chance? I'm just wondering if for some reason it's trying to use Bluetooth rather than WiFi.
> 
> Definitely not looking forward to getting my new iPhone. 🫣

Indeed if anybody can add information on the role of Bluetooth in all this? 
1. Is Bluetooth used for the transfer instead of Wifi?
2. Is there an interference of Bluetooth and Wifi when the two devices are too close?

If that is true, then one solution for both would be to turn Bluetooth off.
If Bluetooth is required for transfer initiation, then 2. could be fixed by keeping devices far apart.
Of course, keeping devices apart denies them the option of a direct Wifi link. Are they trying?

There is so much unknown here and sadly nobody from Signal chiming in on where things stand with this issue?
How is it supposed to work? What is known about why things don't work? What are open questions and development issues? Are there known workarounds for the time being? What is the path to a complete fix?

As for the "complete fix": Improving transfer times by x4 is good. Unfortunately transfer times can be so slow, that x4 means going from crippling slow to abysmally slow. An improvement, yes, but …

**Comment by Jamo33** [comment]
I started the data transfer just now from my old iPhone to the new one.  This is my experience so far:

1. It took 4 attempts before they connected with each other
2. I used the USB-C to Ethernet adapter for the new phone only (not sure if I should have had one for both phones, but the lightning to USB-C is over a week away for delivery for some reason). The resulting speed was 180-200 hours!! I took a screenshot which I will upload when I can get access to my phone again. 
3. I unplugged the ethernet cable and after about 2 minutes of doing nothing the speed changed to 30 hours. 
4. 30 hours is still too long!!
5. There should be an option to select which chats you want to keep and how long within the chat you want to keep.  Most of my group chats I just need a week.  Plenty of others just a few months.  It's only a very rare few that I need to keep all the chats with.

**Comment by neuhaus** [comment]
@Jamo33 have you tried disabling bluetooth?

**Comment by Jamo33** [comment]
> @Jamo33 have you tried disabling bluetooth?

Yes I did that and it didn't change anything

**Comment by neuhaus** [comment]
> > @Jamo33 have you tried disabling bluetooth?
> 
> Yes I did that and it didn't change anything

OK, thanks. So both devices are connected to the same wifi access point? Does the AP permit devices to talk to each other (i.e. is AP client isolation turned off)?

**Comment by Jamo33** [comment]
So after approximately 3 hours this lovely pop up appeared:

![IMG_2417](https://github.com/user-attachments/assets/ad32a998-44b1-4d4b-84fa-3900a53fd44a)

I was using an EERO AP and I couldn't work out how to control the client isolation with it, so I moved rooms to one that has a Netcomm router and that definitely has the client isolation turned off - the wifi anyway - I couldn't see how to turn it off for the ethernet. 

So far though I have only a slightly faster time remaining calculation of 26 hours rather than 30.  That was when I was using the USB-C to Ethernet adapter.  

So I unplugged the adapter just to see what would happen and the time remaining calculation reduced to 24 hours. No idea why that is.

**Comment by Jamo33** [comment]
I managed to get a further time remaining calculation reduction by placing the phones perpendicular to the router.  Previously they were both parallel with one phone closer the router than the other. 

Now it says 14 hours. Going to sleep now!

**Comment by vinnyoflegend** [comment]
@Jamo33 is your bluetooth off? Someone else remarked that it seemed as thought IOS devices are favoring bluetooth ad-hoc transfer. 

Much earlier in this issue thread I found that I had the most success stacking the phones directly on top of each other....

**Comment by Jamo33** [comment]
> @Jamo33 is your bluetooth off? 

Bluetooth is off (it made no difference for me btw)

**Comment by Jamo33** [comment]
I just got to 80% transferred when, for no apparent reason, I got the "transfer couldn't complete" pop-up again.  

So here we go for another 14 hours all over again. :(

**Comment by Jamo33** [comment]
Finally completed the transfer. Weirdly the sending phone had an error message but the receiving phone was ready to go.

**Comment by MalEbenSo** [comment]
> Finally completed the transfer. Weirdly the sending phone had an error message but the receiving phone was ready to go.

Which would leave me with a bad taste that maybe not all data were transferred. 

This whole story reminds me of Boeing at this point. A product that was designed to be trustworthy is losing trust, because key issues (here: migration, backup) are not addressed – not in code and not in communication from Signal representatives about measures and plans to fix this. 

Thanks to @Jamo33 🙏 we have at least even more ideas of what does _not_ help. Or not reliably, because the number of data transfers is small. 

Which, to be clear, doesn’t mean that the number of affected users is small. The issue seems pretty reproducible. I also migrate iPhones for the rest of the family, and the issue reliably appears, only mitigated by smaller chat histories, so I assume affects 100% of iOS users. 
With est. 40 million users and 50% on iOS and devices migrated on average every four years, that is 5 million users per year, or am I missing something? 
Some might not bother to migrate though, in general and then especially because the process is so bad. So let‘s say 1 million per year? 
On the other side the number might be higher, as users migrate their iPads, too. 
And the problem gets worse as chat histories grow.

**Comment by Mr-SourBalls** [comment]
Issue still present, 16GB of signal data, started migrating around 21:00, and it stopped at 88% around 5:30 this morning when my alarm went off so i can restart this entire mess again today. 

From an iPhone 15 Pro Max to an iPhone 15 Pro Max on a plenty fast network (900mbit wireless speeds). 
Combined with the utter lack of good media management makes me want to quit using signal all together.

There is zero way to truly backup my own chat history, no export option nothing. I don't want to lose my chat history with my friends and family. But not using my devices for almost a day to transfer a couple of gigabytes of history is insane.

**Comment by kwaaak** [comment]
Also had a problem when transferring around 20 GiB of data. Was very slow when the phones sat next to each other.

Restarted phones, force closed Signal and re-installed on the new phone.
Set the phones up on the same floor as the WiFi access point (WiFi6 standard, devices both on 5GHz band) but in different rooms and kept them connected to power.

Started the transfer and it went smoothly, I monitored the connection speed which fluctuated between 18 and 22 Mbit/s mostly.
All in all it took around 2.5 hours.

**Comment by CosmicHazel** [comment]
Stopping contributions to Signal until resolved.

**Comment by IskanderNovena** [comment]
Doing a transfer now. Initially, transfer time would be about 7 hours for 20GB+. I've restarted the transfer after connecting the two iPhones using a cable. Transfer time is now 30 minutes for the same data size.
So using a cable greatly improves transfer speeds.

**Comment by Jamo33** [comment]
Well that would have been good to know last month! 

In the mean time, I'm feeling very guilty as my friend who I convinced to get Signal has just discovered that you cannot transfer anything from Android signal to iPhone signal. 

So he loses everything. And he's very annoyed. 😞

**Comment by MalEbenSo** [comment]
We now have two recent indicators that slow transfer speeds might be caused by Signal using Bluetooth instead of faster methods. 

Both plugging in [a cable between the iPhones](https://github.com/signalapp/Signal-iOS/issues/5647#issuecomment-2450213292) and [keeping the iPhones apart and thus effectively disabling Bluetooth](https://github.com/signalapp/Signal-iOS/issues/5647#issuecomment-2418148911) has helped.


@IskanderNovena : 
> Doing a transfer now. Initially, transfer time would be about 7 hours for 20GB+. I've restarted the transfer after connecting the two iPhones using a cable. Transfer time is now 30 minutes for the same data size. So using a cable greatly improves transfer speeds.

What iPhones and cable where you using? Was that USB-C to USB-C or to USB-C to Lightning? 
I would have never thought that connecting two iPhones by cable has any support in iOS. 


@CosmicHazel :
> Stopping contributions to Signal until resolved.

Fwiw, same here. I took this step earlier but silently. Maybe there is a point in noting this issue as the reason.


@Jamo33 :
> In the mean time, I'm feeling very guilty as my friend who I convinced to get Signal 

Fwiw, same here with more than one friend and with family, too. 

If anybody at Signal is listening: I hope you realize that because of data loss issues (slow / failing / impossible transfers, no backups, no export) you are losing the backing of supporters: Financial support and probably worse, recommendations. 

If this does not get fixed, then I expect that Signal's reputation as a reliable messenger platform will erode, and so will the user base.

**Comment by ast3150** [comment]
Initially, Signal displayed a transfer time of multiple hours. I force-quit the app on both old and new phone, disabled Bluetooth on both, and started the transfer again. The estimated time dropped to about 15 minutes using WiFi 6E.

This is a bad UX and needs to be addressed urgently by the Signal devs.

Also I didn't realize that Signal makes no backups whatsoever. I'm now worried about losing all my chat history and saved media if I ever lose my phone or it gets stolen.

**Comment by IskanderNovena** [comment]
@MalEbenSo 
> What iPhones and cable where you using? Was that USB-C to USB-C or to USB-C to Lightning?
> I would have never thought that connecting two iPhones by cable has any support in iOS.

Apple mentions connecting phones through a cable when using Quick Start to transfer settings between the new and old iPhone: https://support.apple.com/en-us/117383

I connected using an USB-C to Lightning cable (since I'm coming from an iPhone 13 mini). But it should work with USB-C to USB-C as well.

**Comment by Jamo33** [comment]
The team at Signal should let EVERYONE who uses the app know they can use a cable to transfer the files.  It should be mentioined on the transfer process "make sure both screens to not turn off and plug the iphones in using usb-c / lightning cables for the fastest transfer experience".

**Comment by Doc1faux** [comment]
Thanks @MalEbenSo for the tip!
This should definitely be added to the [iOS Backup and Restore guide](https://support.signal.org/hc/en-us/articles/360007059752-Backup-and-Restore-Messages#ios_restore).
Firstly I strictly followed the guide and seen an estimated time of 3h and a very slow progress bar... 
Searching the Web for this issue, I found this GitHub one and decided to abort the transfer, kill both apps, disable BLE, linking both phone via USB/lightning and start again and oh god, the transfer took 5mns!

**Comment by lankytom** [comment]
Transfer of 8GB from iPhone 15 to 16 was at 30% after 2 hours, with 6 (!) more hours to go.
So I arrived here at this thread, looking for solutions.
I cancelled the transfer, turned BT and Airdrop off, connected the devices USB-C to USB-C, and the transfer took 8 minutes to complete. Great!

However, the wireless transfer issue should get fixed, or at least there should be a hint about using a cable will speed up the process.
And, a backup/restore feature is needed too, imho, I'm totally relying on Signal with most of my comms, and not having a backup, in addition to these basically unresolved transfer issues makes me feel nervous.

**Comment by ameesme** [comment]
I can confirm the above strategy to work perfectly when using USB-C to Lightning as well. Just transferred from my iPhone 12 Pro to my iPhone 16 Pro in 16 minutes instead of three hours🤯

**Comment by Squiddim** [comment]
Hello,

i saw the Post about the new "Synchronization Progress" which is now in Beta and the post mentioned:

"First, your primary device compresses all message history into a single transferable archive" ~  #https://signal.org/blog/a-synchronized-start-for-linked-devices/
This sounds like most of the work for a Backup and Restore feature on IOS is already there?

Seemed relevant to this issue

**Comment by Jaffex** [comment]
This is one of those issues that are incredibly frustrating for non-tech users that want to keep their message history without much hassle.

Just had a friend of mine being incredibly frustrated and wanting to quit signal altogether after being confronted with this issue.

In their case, the transfer aborted after them waiting for almost an hour for it to complete.

Seeing that the transfer would then begin anew, they just decided to abandon their old messages and just being frustrated about signal as a whole.

**Comment by ryanmcgrath** [comment]
I usually don't `+1` a random GitHub issue, but this one is particularly frustrating and I can't help myself. Considering the usage numbers that iOS has in the west, I find it surprising that this is still an issue.

I'm sitting in an Apple store at the moment waiting on the most painfully slow transfer between my old phone and a replacement phone. The cord/port route is not an option due to port failure (the reason for the phone replacement). I'm grateful for Signal's existence and I find myself advocating for it in numerous groups I'm in, but I consider this a glaring downside that should be prioritized.

(I find that my experience is pretty similar to [notpeter](https://github.com/signalapp/Signal-iOS/issues/5647#issuecomment-1955090777)'s experience further up the chain)

Can someone at Signal acknowledge this and/or throw out some kind of update on whether this is being looked at?

**Comment by MalEbenSo** [comment]
> I'm sitting in an Apple store at the moment waiting on the most painfully slow transfer between my old phone and a replacement phone. The cord/port route is not an option due to port failure (the reason for the phone replacement). […] 
> _and_, just to be clear, my total data to move is seemingly only ~4GB. This is less than others here but being forced to bluetooth just cripples the process.

Since the cable connection is not an option, did you try to disable Bluetooth and thus force Signal to use WiFi? This seems to have worked for others.

Note: My comment is by no means intended to justify Signal's painful migration issues in the sense of "you're migrating it wrong". 

@Signal: Please fix this. 
And the fix might be simple enough. I appears that by default Signal chooses a slow data connection via Bluetooth when Signal should instead choose faster connections when available (usually Wifi) and inform users, that an even faster connection (cable) is possible.

**Comment by ryanmcgrath** [comment]
> Since the cable connection is not an option, did you try to disable Bluetooth and thus force Signal to use WiFi?

This unfortunately does not work when you're sitting in an Apple store due to isolation configuration on their wifi networks (someone else further up on the thread also noted this).

> I appears that by default Signal chooses a slow data connection via Bluetooth when Signal should instead choose faster connections when available (usually Wifi) 

Reading through the code, Signal's currently just using Apple's MPC framework. That, to my knowledge, doesn't give you a way to "prefer" a connection - and it goes in descending order with Bluetooth being the last.

> inform users, that an even faster connection (cable) is possible.

This unfortunately wouldn't help those of us doing transfers on SE devices, but those devices are on their way out anyway. For USB-C capable devices this should _100%_ be noted.

**Comment by antoszka** [comment]
I can confirm this is still an issue — both for the current public and beta (via TestFlight) releases.

I have a considerable history of Signal conversations, and after 8 hours of transferring I have the progress bar at 56% and I'm "praying" nobody is gonna call or any other disturbance will briefly bring Signal to the background causing the whole night of transferring to waste. This is so unimaginably badly designed... I will try the workarounds suggested here and report back if I end up needing to retry.

**Comment by JoshuaSchlichting** [comment]
This is still a problem March 2025. Finding this GitHub issue, I was able to get this from 8 hours down to 1 hour by force closing the app in both phones, turning off bluetooth, and starting over. 

I did try to use a cable, as suggested here, but the later model iPhone threw some sort of error about the older iPhone being an accessory that was drawing too much power (that error would spam me as the older iPhone constantly went into a charging state, on and off).

**Comment by neuhaus** [comment]
I managed to get the data transferred fairly quickly a few days ago between an iPhone 13 and an iPhone 16 Pro using a lightning-to-USB type C cable.
The transfer took less than half an hour for 8GB or so. 
P.S. I still want to be able to export my own data!

**Comment by costyn** [comment]
As a fairly new Signal user, I only had to transfer 225MB, and it still took 15 minutes!

**Comment by D300** [comment]
at least on desktop app macos it merges the data

**Comment by cpatulea** [comment]
iPhone SE to iPhone SE (2022) trying to transfer ~4.4 GB.

Tried with default settings, barely runs, let it run overnight, got stuck at around 34%. Tried to disable Bluetooth (to force wifi), barely any change, still impossible to complete the transfer.

Trying via wire, using Apple Lightning to USB Camera Adapter + ordinary USB to lightning cable. Transfer worked, took about 30 minutes for the 4.4 GB.

**Comment by Jamo33** [comment]
I really do not understand why this isn’t in the instructions for
transfers. It’s not hard to do and it will save a lot of people a lot of
angst.

**Comment by MADXhh** [comment]
This is still a problem!!!! 
The tip about the cable was a game changer!! I'm switching from iPhone 13 to 16, and thanks to a USB-C to Lightning cable, it took just minutes. I had tried it before without a cable and gave up after more than 3 hours.

**-> close the app (both phones), turning off bluetooth & wifi (airplane mode), and starting over.**

**Comment by antoszka** [comment]
I really do not understand why this hasn't been decently fixed for the past n-years. Tens of millions of dollars in funding and we get... stickers. People still have to hunt around forums, github and pick random advice that requires a possibly exotic cable, disabling all wireless comms on your phones — it might be a little over-the-top for a non-nerdy user.

**Comment by pirate** [comment]
I've now lost my entire Signal mesasge history on iOS 3 times over the last 8 years due to buggy device<->device transfers or failed iOS upgrades.

Entire chapters of my life wiped out with no recourse, incredibly frustrating and frankly unforgivable for a decade+ old platform at this point. I regret donating and recommending my family members switch over because data loss has been such a pervasive issue on iOS and there is no solution.

Adding stickers, stories, and $ payments in-app before a real backups solution is just salt in the wound. Please please please let us back up our iOS data and fix iOS transfers. We don't even need cloud backups, just give me some buried menu option to export/import an encrypted sqlite file.

**Comment by MalEbenSo** [comment]
> I really do not understand why this isn’t in the instructions for
> transfers. It’s not hard to do and it will save a lot of people a lot of
> angst.
> […](#)

It will not just save them from angst, but also from data loss, as others here have posted.

As @pirate just [wrote](https://github.com/signalapp/Signal-iOS/issues/5647#issuecomment-3240747800): The transfer problem is connected wrt user experience with the lack of backup options. If the transfer fails and there is no backup, then the data are gone. 

I am ok for myself with cable for transfers and sigtop for backups. But I am not happy. The user experience is just not good.
And I have a bad conscience recommending signal, which I did in earlier days and with good reception from others. 
I have also stopped donating, because like @pirate I don't understand the priorities for using the money.

**Comment by neuhaus** [comment]
Yes eventually we will have real local backups, but it sounds as if we will have to wait quite a bit longer before it arrives.

Also I'm of the opinion that we should be allowed to decrypt our own local data on our own computer if we want. I don't see this possibility in the Signal roadmap.

**Comment by cars11** [comment]
Ridiculous. Using signal for so many years and this is still an issue. I even have WiFi 7 now and it still wants 3 hours. 

Not donating to signal anymore until this is sorted. 
Also local encrypted backups please

**Comment by N9PBJ** [comment]
This alone had me seriously thinking of giving up on signal. 
Over 2 hours to transfer my history. Finally found the tip on using a cable, and it took 7 minutes.

Lightning to USB adapter cable on old iphone, then a USB C to USB A on new iphone 17. Turned off bluetooth and wifi just in case on both phones.

**Comment by pfych** [comment]
It would be awesome to be able to choose what to transfer across, sitting here with a cable plugged in to both phones waiting with "About 1 hour remaining" for a while now. Frustrating

**Comment by lankytom** [comment]
I migrated my Signal from an iphone 16 Pro to a 17 yesterday, by cable (so USB-C fast to USB-C slow), and it took 11 Minutes. This is acceptable. Switching to flight mode is very important, or else if someone calls while the transfer is in progress, it will interrupt and can't resume. So you will have to start over from the beginning. As happened to me yesterday :-(

**Comment by sneakerdad** [comment]
My iPhone 17 Pro arrived today (upgrading from 16 Pro) and finally, for a
change, I successfully transferred 1GB in roughly 4 minutes. This was all
thanks to the advice on this thread.

Here’s what I did:
-airdrop disabled on both phones
-airplane mode activated on both phones
-WiFi disabled on both phones
-Bluetooth disabled on both phones
-Signal app forced-quit on both phones
-phones connected by Thunderbolt 4 cable

Only upon this setup did I open the Signal apps on both phones.

Hoping this post helps someone else (and maybe makes its way into the
Signal apps UI)

**Comment by The-May** [comment]
Transfer gets dramatically better when disabling bluetooth and airdrop.
Then I both connected my 11 and 17 with an ethernet Adapter and it went from 13Hours to 30 Minutes. Fyi

**Comment by lankytom** [comment]
> Transfer gets dramatically better when disabling bluetooth and airdrop. Then I both connected my 11 and 17 with an ethernet Adapter and it went from 13Hours to 30 Minutes. Fyi

Ethernet ??

**Comment by El-Jefaso** [comment]
How are people using a cable? I have iPhone 13 Pro Max(old) connected to iPhone 17 Pro Max(new), both phones in airplane, but the new phone asks for sms verification to continue.

**Comment by N9PBJ** [comment]
> How are people using a cable? I have iPhone 13 Pro Max(old) connected to iPhone 17 Pro Max(new), both phones in airplane, but the new phone asks for sms verification to continue.

I had to use the Lightning to USB adapter on my iphone 14, then just the regular usb c to usb a cable that came with the iphone 17.

**Comment by El-Jefaso** [comment]
> > How are people using a cable? I have iPhone 13 Pro Max(old) connected to iPhone 17 Pro Max(new), both phones in airplane, but the new phone asks for sms verification to continue.
> 
> I had to use the Lightning to USB adapter on my iphone 14, then just the regular usb c to usb a cable that came with the iphone 17.

i have an apple usb-c to lighting cable between the two. what adapter are you talking about? ty

**Comment by max-signal** [comment]
The existing device-to-device transfer feature is built on the [Multipeer Connectivity framework](https://developer.apple.com/documentation/multipeerconnectivity). As evidenced by this issue, there are either problems with our implementation or we’re pushing up against the limits of the framework. From the documentation:

> In iOS, the framework uses infrastructure Wi-Fi networks, peer-to-peer Wi-Fi, and Bluetooth personal area networks for the underlying transport.

As far as I know, we can’t explicitly specify which underlying transport mechanism we want to use, so the workarounds here (disabling Wi-Fi/Bluetooth, connecting a cable, etc.) are likely the best way to force the device to pick the fastest transport mechanism. (If anyone knows how to force a faster transport mechanism, please let me know.)

**Comment by neuhaus** [comment]
@max-signal If I remember correctly, the app currently asks you to enable bluetooth, instead there should be an added text to give clues how to optimize the transfer rate. That‘s an easy solution to improve the situation.

A better solution could be to migrate to another mechanism, perhaps following the description at https://developer.apple.com/forums/thread/776069

**Comment by qlp** [comment]
> The existing device-to-device transfer feature is built on the [Multipeer Connectivity framework](https://developer.apple.com/documentation/multipeerconnectivity). As evidenced by this issue, there are either problems with our implementation or we’re pushing up against the limits of the framework. From the documentation:

Seems multipeer connectivity is not a good fit for big transfers? Maybe consider migrating to [Network Framework](https://developer.apple.com/documentation/Network) using [peer to peer](https://developer.apple.com/documentation/network/nwparameters/includepeertopeer)?

(no expert in this field; just been looking for alternatives)

**Comment by auchenberg** [comment]
Came here after upgrading to the iPhone 17 Pro: 

To enable super fast transfers: 
1. Disable wifi
2. Plug in the USB-C cable between your new iPhones.
3. Start transfer process

**Comment by directionforward** [comment]
Confirming @auchenberg solution to force a cable transfer.

1. Quit Signal on both new and old devices.
2. Turn off Wifi, you can leave Bluetooth on. Also turn off AirDrop.
3. Then connect via cable.

My transfer went down from 5 hrs projected to less than 10 mins.

**Comment by ejade** [comment]
> Came here after upgrading to the iPhone 17 Pro:
> 
> To enable super fast transfers:
> 
>     1. Disable wifi
> 
>     2. Plug in the USB-C cable between your new iPhones.
> 
>     3. Start transfer process

I disabled wifi and bluetooth on one device (receiving one) and plugged in cable and started transfer process again. It completed in about 15 minutes without any errors.

**Comment by louie337** [comment]
> > Came here after upgrading to the iPhone 17 Pro:
> > To enable super fast transfers:
> > ```
> > 1. Disable wifi
> > 
> > 2. Plug in the USB-C cable between your new iPhones.
> > 
> > 3. Start transfer process
> > ```
> 
> I disabled wifi and bluetooth on one device (receiving one) and plugged in cable and started transfer process again. It completed in about 15 minutes without any errors.

This disabling WiFi from the receiving device with cable worked for me. 
I transferred my data from iPhone SE (lightning port) to iPhone 17 with a lightning - USBC cable.
Transfer speed was improved drastically

**Comment by MattiBug** [comment]
> > Came here after upgrading to the iPhone 17 Pro:
> > To enable super fast transfers:
> > ```
> > 1. Disable wifi
> > 
> > 2. Plug in the USB-C cable between your new iPhones.
> > 
> > 3. Start transfer process
> > ```
> 
> I disabled wifi and bluetooth on one device (receiving one) and plugged in cable and started transfer process again. It completed in about 15 minutes without any errors.

this worked for me. iphone 13pro to 17pro. 10 minutes with cable, 7+ hours without... crazy bug

**Comment by sebastius** [comment]
Ridiciously slow transfer from iPhone 12 mini to iPhone 17, both on IOS26.0.1. Fixed it by plugging in a cable. I wish the app would just say it was an option to use the cable.

**Comment by amadeuszxd** [comment]
Please add this information to app. I just came from whole night and morning migration, about 13 hours from 12 pro to 17 pro max.

**Comment by tschibu** [comment]
@amadeuszxd  - that 100% true - I saw this thread too late and then unfortunately decided to shoot the Signal data into the moon - now I have lost 4 years of chats.. I will slowly try to get away from the Signal - unfortunately

**Comment by le-rou** [comment]
Thank you! I also had this problem - It went well going from iPhone 12 mini to iPhone 17 (~2 hours) but a week later from iPhone 17 to another iPhone 17, it aborted several times. I switched to LAN via USB-C adapters to get more speed and stability: 60% after 6 hours. Did not know about the USB-C cable direct connection. This should be told in the Signal App! I disabled WIFI, plugged in a cable and started the progress. It took less than 10 minutes. Thanks!

**Comment by VividVisions** [comment]
Thank you, @auchenberg ! Wi-Fi: 6 hours. Cable: 15 minutes. 
How is this still a problem, more than two years later?

**Comment by mohoromitch** [comment]
> Came here after upgrading to the iPhone 17 Pro:
> 
> To enable super fast transfers:
> 
> 1. Disable wifi
> 2. Plug in the USB-C cable between your new iPhones.
> 3. Start transfer process

Adding some more data points to train the LLMs.

Transferring from an iPhone 13 Pro to an iPhone 17 Pro, both on iOS 26.0.1.

I tried three times wirelessly. The first time, I was around 60% when I accidentally tapped a notification. 😢 The second time, it took about an hour and got to 99% and 0 minutes remaining, and was stuck there for >30 minutes before I killed it. Both of these attempts were done at home, were estimated to take **around an hour.** 

I tried a third time at the office, but the estimated time was **five hours**, so I tried to find another way and that led me to this issue.

I had a USB-A to Lightning cable, and used a USB-A to USB-C adapter to connect the two phones together. The estimated transfer time dropped to **~15 minutes**, and successfully completed. 🎉

I wish that this was mentioned in the UI, it would've saved me a couple hours or so of time—and practically everyone has the required cable to do so.

**Comment by hermanho** [comment]
> Confirming [@auchenberg](https://github.com/auchenberg) solution to force a cable transfer.
> 
> 1. Quit Signal on both new and old devices.
> 2. Turn off Wifi, you can leave Bluetooth on. Also turn off AirDrop.
> 3. Then connect via cable.
> 
> My transfer went down from 5 hrs projected to less than 10 mins.

It works for me.
Transfer from iPhone 16 Pro to iPhone 17 Pro. The time is reduced from 1 hour to 2 minutes.

**Comment by phil-davis** [comment]
Transferring from an iPhone11 to another iPhone11. I got the "hangs at 99%" problem.
Restarted both phones. Started the transfer again. Put the phones "head to head" sitting flat on a surface (both face up), and it worked. No idea if this is a "solution" that is reliable.

**Comment by FlorianWendelborn** [comment]
Ran into it getting stuck at 100% on iPhone 13 Pro Max -> iPhone 17 Pro Max

Workaround: retry after force quitting both apps on both phones

**Comment by arndjan** [comment]
As it stands, I would settle for slow. I can’t get the transfer to start at all. Have tried everything. Transferring from an iPhone 8 Plus to a 13 Pro. As soon as it scans the QR code, the old iPhone freezes and nothing ever happens after that.

**Comment by MattiBug** [comment]
> As it stands, I would settle for slow. I can’t get the transfer to start at all. Have tried everything. Transferring from an iPhone 8 Plus to a 13 Pro. As soon as it scans the QR code, the old iPhone freezes and nothing ever happens after that.

use a cable. it is the way!

**Comment by davetempleton** [comment]
I tried using the advice to turn on Airplane Mode and turn off wifi/bluetooth/airdrop, and that DIDN'T work for me. The signal app wouldn't generate the QR code without any connectivity, and the other phone wouldn't want to connect with no network enabled.

I'm setting up a new phone where I just transferred the SIM/eSIM from one to the other, so only one phone (the new phone) has a cellular connection. This is what worked for me:

NEW PHONE (with working cellular): Air Drop Receiving Off, Airplane mode Off, Cellular Data On, Wifi Off, Bluetooth Off

OLD PHONE (no cellular, just "SOS"): Air Drop Receiving Off, Airplane mode Off, WiFi On (and connected to my network with internet access), Bluetooth Off

Force quit Signal on both phones. Connect a USB-C to USB-C cable (this may work with other cable combinations described in other posts if you have one phone with Lightning). Start Signal on both apps and try to do the transfer. Should generate the QR code, allow you to the the picture, and then start transferring. I'm doing about 10GB in 15 min.

**Comment by davetempleton** [comment]
> As far as I know, we can’t explicitly specify which underlying transport mechanism we want to use, so the workarounds here (disabling Wi-Fi/Bluetooth, connecting a cable, etc.) are likely the best way to force the device to pick the fastest transport mechanism. (If anyone knows how to force a faster transport mechanism, please let me know.)

Max, my sincere recommendation is that you change the UI and suggest that people pursuing this method use USB cables and disable wireless adapters. Your team will probably need to do tests to see what works for different aged devices, connecting various devices with Lightning vs. USB-C, etc.

But I promise you, you need to have a "there be dragons" message to users who insist on trying to do wireless transfer, especially if their storage is >1GB. No one should have the experience I just had, and then had to hunt down this thread and read 80 replies before I found someone describing the right way to use a cable, and that not working for me and having to troubleshoot further to get it to work. I'm a professional IT person like many here, and this totally ruined an entire Friday evening dealing with this.

**Comment by akaupang** [comment]
> I haven't confirmed this, but I think the issue may be exacerbated by network conditions. [...]
> 
> Hilariously I was able to work around this by using a wireless hotspot. I connected both phones to the wireless hotspot and instead of the many hour estimate I got on the Apple Wifi everything transferred in ~40mins.

I'd say @notpeter is on to something, for those who are unable to use a cable (I didn't try).

I experienced slow transfers and transfer failures while migrating for a friend from iPhone 11 to 17 with Signal 7.89 on both phones. The transfers were estimated to take hours and failed a few times on the home network (a simple eero router). Phones at 3 m from router and 40 cm apart. Bluetooth on (both phones). Reboot both and retry with no change.

Curiously, the migration went smoothly and finished in about 45 minutes when both iPhones were connected to an Android 14 hotspot (Fairphone 4, with iPhones 40 cm to each side). I have no idea as to the amount of data the transfer entailed (5 years of daily media-rich conversation with a single recipient).

In any case, hope this helps someone else.

**Comment by jurajlutter** [comment]
I have spent more than an hour trying to figure out what's the problem with the transfer. Tried cables, hotspots, wifi, airplane mode, this and that, everythig. Then I've stumbled upon on an older reddit article that pointed me to the right direction.

ALWAYS CHECK WHETHER YOU ALLOWED SIGNAL TO ACCESS THE LOCAL NETWORK (Settings -> Apps -> Signal -> Local netwsork). It was turned off on the old phone. After enabling, the transfer is working.

**Comment by redm123** [comment]
Just stumbled over this, coming from a database corruption due to out of storage space, with complete data loss. I had this pain, as well, transferring to my current phone a while ago. Reading this thread one could just lough, again, if it weren't so sad. This mechanism seems incredibly make shift and brittle. 

And it doesn't help you, if you don't have the original phone, because it broke, got stolen or whatever. You want to migrate between Android and iOS, or whatever. And you even may have all the data right there, in your Signal Desktop for instance. But you cannot use it! Broken by design, I would say. 

They are so obsessed with privacy and security, they forget or don't care (I suspect) that some (most) people also just want to use the software and actually also care about their data. They happily accept user data loss (shrug), or declare it a feature... In addition every variant of Signal does its own thing here.

And the solution to all this would be so obvious: a device independent backup! Where I can create and store locally, on my encrypted storage at home, and replay if necessary to any flavor of Signal apps.

Yea, they started something now with these Secure Backups. Finally! But their cloud only (haha!), limited, and I still could not restore from one device to another. Yea, their blog reads like they plan this, but with the attention this got up to now, we can probably check back in 10 years and many data losses later... 
Sorry for the somewhat frustrated rant..

**Comment by undeadclicker** [comment]
Thanks to everyone for the helpful suggestions much like most I started with attempting the wireless backup which was estimated at 4 hours and failed after I walked away from the phones and they went to sleep. After that I did some googling and found this thread. I able to get the transfer down to 12 minutes by connecting a lightning to usb a cable to a usbc to usba adapter going from iphone xr to iphone 17. I turned on wifi on the old phone and turned off wifi on the new iphone. Even after all that the first few attempts failed, but I closed and relaunched the app and tried again until I got it to finally transfer. Thanks @auchenberg and @mohoromitch for the info!
